notifier/notifier.go

```diff
@@ -298,42 +301,101 @@ func (n *Manager) nextBatch() []*Alert {
 	return alerts
 }
 
+// Run dispatches notifications continuously, returning once Stop has been called and all
+// pending notifications have been drained from the queue (if draining is enabled).
+//
+// Dispatching of notifications occurs in parallel to processing target updates to avoid one starving the other.
+// Refer to https://github.com/prometheus/prometheus/issues/13676 for more details.
+func (n *Manager) Run(tsets <-chan map[string][]*targetgroup.Group) {
+	defer n.cancel()
```

Comment:
Why do we still need to run the notifier through a cancellable context? After this PR changes, the notifier stop will always "block" until an in-flight notification has done.

Reply:
Good catch, we can remove this. Removed in 5c19890.